### PR TITLE
Upgrade deployment to cass-operator v1.20.0

### PR DESCRIPTION
**Pull request by burmanm**
**What this PR does**:
Update to cass-operator v1.20.0

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: config/cass-operator/ns-scoped/kustomization.yaml [hunk 1]
```diff
@@ -2,11 +2,4 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 
 resources:
-- github.com/k8ssandra/cass-operator/config/deployments/default?ref=v1.19.1
-
-components:
-  - ../../components/cass-operator-image-config
```
Comment (adejanovski) [inline config/cass-operator/ns-scoped/kustomization.yaml:8]: **Question:** Why was this removed?

File: config/components/cass-operator-image-config/image_config.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ kind: ImageConfig
 metadata:
   name: image-config
 images:
-  system-logger: "k8ssandra/system-logger:v1.19.1"
+  system-logger: "k8ssandra/system-logger:v1.20.0"
   config-builder: "datastax/cass-config-builder:1.0.7"
   k8ssandra-client: "k8ssandra/k8ssandra-client:v0.2.2"
```
Comment (adejanovski) [inline config/components/cass-operator-image-config/image_config.yaml:8]: **Todo:** shouldn't k8ssandra-client be upgraded to `v0.3.0`?

File: config/cass-operator/ns-scoped/kustomization.yaml (hunk 1 above)
Comment (burmanm) [reply config/cass-operator/ns-scoped/kustomization.yaml:8]: Because it has no purpose?

**Review by adejanovski — APPROVED** [review]
